src/price_data.py

```diff
@@ -414,3 +419,163 @@ def get_cost(
         if isinstance(tr, transaction.SoldCoin):
             return price * tr.sold
         raise NotImplementedError
+
+    def get_candles(self, start: int, stop: int, symbol: str, exchange: str) -> list:
+        exchange_class = getattr(ccxt, exchange)
+        exchange_obj = exchange_class()
+        if exchange_obj.has["fetchOHLCV"]:
+            time.sleep(exchange_obj.rateLimit / 1000)  # time.sleep wants seconds
+            # get 2min before and after range
+            startval = start - 1000 * 60 * 2
+            rang = max(int((stop - start) / 1000 / 60) + 2, 1)
+            return list(exchange_obj.fetch_ohlcv(symbol, "1m", startval, rang))
+        else:
+            log.error("fetchOHLCV not implemented on exchange, skipping ohlcv")
+            # shouldnt happen technically because exchanges are filterd for fetchohlcv
+            return []
+
+    def _get_bulk_pair_data_path(
+        self,
+        operations: list,
+        coin: str,
+        reference_coin: str,
+        preferredexchange: str = "binance",
+    ) -> list:
+        def merge_prices(a: list, b: list = []) -> list:
+            prices = []
+            if not b:
+                return a
+            for i in a:
+                factor = None
+                for j in b:
+                    if i[0] == j[0]:
+                        factor = j[1]
+                        break
+                prices.append((i[0], i[1] * factor))
+            return prices
+
+        timestamps: list = []
+        timestamppairs: list = []
+        maxminutes = (
+            300  # coinbasepro only allows a max of 300 minutes need a better solution
+        )
+        timestamps = [op.utc_time for op in operations]
```

Comment:
Maybe find a way to improve this.
e.g for binance request the whole 1000candles and for the rest resplit them into smaller chunks according to the size